controllers/bounce_processes.go

```diff
@@ -219,8 +223,10 @@ func getProcessesReadyForRestart(logger logr.Logger, cluster *fdbv1beta2.Foundat
 	// if some processes are already upgraded e.g. in the case of version compatible upgrades and we also don't want to
 	// block the restart command if a process is missing longer than the specified GetIgnoreMissingProcessesSeconds.
 	// Those checks should ensure we only run the restart command if all processes that have to be restarted and are connected
-	// to cluster are ready to be restarted.
-	if cluster.IsBeingUpgradedWithVersionIncompatibleVersion() && (counts.Total()-missingProcesses) != len(addresses) {
+	// to cluster are ready to be restarted. If more than one storage server per Pod is running we have to account for this.
+	expectedProcesses := counts.Total() - missingProcesses + (counts.Storage * (cluster.Spec.StorageServersPerPod - 1))
```

Comment:
Is it better to modify "cluster.GetProcessCountsWithDefaults()" to take "cluster.Spec.StorageServersPerPod" into account, rather than doing this adjustment on the invoker's side? And, have we checked the other invoker functions of "cluster.GetProcessCountsWithDefaults()" to see if they need to do a similar adjustment?

Reply:
Not really, the difference here is that the process counts in most cases reflect the number of Pods and not the processes (the naming here is not really great) and in most cases the operator is only working on Pods as a whole without taking into account how many processes are running inside that Pod.

Reply:
Ok. (A comment above "cluster.GetProcessCountsWithDefaults()" that it doesn't take "cluster.Spec.StorageServersPerPod" would be nice, so other readers would be aware of this detail.)

Reply:
Good point 👍 Let me add a comment on the method.